app/scripts/reducers/cumulus-instance.js

```diff
@@ -1,25 +1,25 @@
 'use strict';
-import { set } from 'object-path';
 
-import {
-  ADD_INSTANCE_META
-} from '../actions/types';
+import {get} from 'object-path';
+import {ADD_INSTANCE_META} from '../actions/types';
+// In Node 12.0.0+, use built-in Object.fromEntries instead of fromPairs
+import fromPairs from 'lodash.frompairs';
 
 export const initialState = {};
 
 export default function reducer (state = initialState, action) {
-  state = { ...state };
   const { data } = action;
-  switch (action.type) {
-    case ADD_INSTANCE_META:
-      if (data.cmr) {
-        if (data.cmr.environment) set(state, 'cmrEnvironment', data.cmr.environment);
-        if (data.cmr.provider) set(state, 'cmrProvider', data.cmr.provider);
-      }
-      if (data.cumulus) {
-        if (data.cumulus.stackName) set(state, 'stackName', data.cumulus.stackName);
-      }
-      break;
+
+  if (action.type === ADD_INSTANCE_META) {
+    state = {
+      ...state,
+      ...fromPairs([
+        ['cmrEnvironment', get(data, 'cmr.environment')],
+        ['cmrProvider', get(data, 'cmr.provider')],
+        ['stackName', get(data, 'cumulus.stackName')]
+      ].filter(([_, value]) => value))
+    };
```

Comment:
This isn't blocking, but usually reducers have to handle multiple actions, hence the switch statements. Of course you can achieve the same thing with if statements, but it's sort of idiomatic to use switch and that's how the rest of the code in the dashboard is written. I'd prefer the consistency for the next person who might need to handle another action here in this reducer, but it's not critical

Reply:
Yep, makes sense. Since there was only 1 condition, I made it an `if` because the IDE was telling me it was silly to use `switch` in this case, and I like to eliminate as many warnings in my IDE as I can.